trie/trie.go

```diff
@@ -31,6 +31,28 @@ type Trie struct {
 	sha      hash.Hash // hash calc for trie
 }
 
+// CopyTrie returns a copy of the given trie.
+func CopyTrie(t *Trie) (*Trie, error) {
```

Comment:
recommend to use a receiver so that the copy function becomes a method of Trie :
func (t *Trie) Copy() (*Trie)

Reply:
Hi sufay, we use reviewable tool to review code. please click here to start review.
![image](https://user-images.githubusercontent.com/4971353/39557324-47c447a6-4eb9-11e8-925a-8994f259568f.png)